pstore/api.go

```diff
@@ -34,11 +37,108 @@ type Record struct {
 	Timestamp time.Time
 }
 
+// Writer is the interface for writing to persistent store.
+// Implementations of Writer must be safe to use with multiple goroutines.
 type Writer interface {
+	// Write writes given collection of records to persistent storage
+	Write(records []Record) error
+}
```

Comment:
I think it may be confusing to re-use the standard Writer interface name for a different interface. I have a preconception that "everybody knows" that a Writer interface is io.Writer. What about PStoreWriter or RecordWriter or something?